src/gui/transferlistsortmodel.cpp

```diff
@@ -159,19 +165,39 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
             const auto activeL = leftValue.toInt();
             const auto activeR = rightValue.toInt();
             if (activeL != activeR)
-                return activeL < activeR;
+                return activeL - activeR;
```

Comment:
won't it overflow?

Reply:
they are int and always be >= 0

Reply:
Why not use `threeWayCompare()` here? (And all other similar places)

Reply:
It's more optimal

Reply:
https://github.com/qbittorrent/qBittorrent/pull/14423#discussion_r580464725

Reply:
> Why not use `threeWayCompare()` here? (And all other similar places)
> #14423 (comment)

`left - right` is more cooler than `threeWayCompare` 😎

Reply:
You rather risk having overflow just to save some minuscule performance? I don't agree here...

Reply:
>>they are int and always be >= 0
>
>Then you should make sure this is guaranteed to be true than just some comment on github.

If this data should be non-negative by design, then we can just add some assertion. Otherwise, we can hide the error from another part of the code.
This is actually quite an interesting problem. I often come across such a pseudo-safe approach in existing code, which can be a source of elusive logical errors.

Reply:
> I pray you don't code for life-critical systems...

![image](https://user-images.githubusercontent.com/34717789/108867195-f36a7680-761a-11eb-8a5e-973af5fd95e0.png)

in my defense, it's known that they are always >= 0 and int, so subtraction would never go out of the domain of int

Reply:
>If this data should be non-negative by design, then we can just add some assertion.

Exactly, this is what I'm implying. And using `threeWayCompare()` would save these assertions.

>This is actually quite an interesting problem. I often come across such a pseudo-safe approach in existing code, which can be a source of elusive logical errors.

Rule of thumb: use comparison operators when determining orders, not arithmetic operators.

Food for thoughts: the three way comparison operator in C++20 does not use a number for the result but a specific type.

Reply:
>Rule of thumb: use comparison operators when determining orders, not arithmetic operators.

At least I wouldn't mix it at this level. As I said above, if such an optimization makes sense it should be placed inside `threeWayCompare()`.

As for the "pseudo-safe approach" I mentioned earlier, you probably didn't understand what I meant. However, I do not want to continue with this topic here, so as not to slow down this PR.